### PR TITLE
feat: add working help overlay

### PR DESCRIPTION
**Pull request by SSelinn**
## Description

This branch adds a help canvas that shows the keybinds and their actions. These are not configurable as it is an image and a png.

## Testability

You can go to the scene _Settings_ or _WinCondition_.
From the _Settings_  scene you can go from the menu go to settings and then to keybinds and back. You can also do this with F1, but it wil take you back to the UI state where from where you left
This is also the case when in game.

## Checklist
- [x] Set the proper pull request name
- [x] Merged main into your branch
- [ ] Added a scene to the game for your changes

### CODE REVIEW

File: aplib.net-demo/Assets/Art/UI/Background/HelpScreen.png
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Art/UI/Background/HelpScreen.png:1]: You forgot to add the jump key :)
Also you can press left alt to make the cursor show up at any time

File: aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs [hunk 1]
```diff
@@ -40,13 +40,13 @@ public class CanvasManager : MonoBehaviour
     /// Name of the start screen.
     /// </summary>
     [SerializeField]
-    private string _sceneNameStartingMenu = "Settings"; // TODO:: Load main start screen
+    private const string _sceneNameStartingMenu = "Settings"; // TODO:: Load main start screen
```
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs:43]: I would make them readonly instead of const, now you cannot edit them in the editor anymore

```suggestion
    private readonly string _sceneNameStartingMenu = "Settings";
```

File: aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs [hunk 2]
```diff
@@ -40,13 +40,13 @@ public class CanvasManager : MonoBehaviour
...
 
     /// <summary>
     /// Name of the game scene
     /// </summary>
     [SerializeField]
-    private string _sceneNameGame = "InGameSettings"; // TODO:: Load main game screen
+    private const string _sceneNameGame = "WinCondition"; // TODO:: Load main game screen
```
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs:49]: TODO's can be removed, I'd set the game scene default to 'GridSystem' now
```suggestion
    private readonly string _sceneNameGame = "WinCondition";
```

File: aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs [hunk 3]
```diff
@@ -82,27 +82,28 @@ private void Awake()
...
+                MenuCanvas.SetActive(true);
+                break;
+            case _sceneNameGame:    // Are we at the main gaming scene?
+                HealthComponent playerHealth = GameObject.FindGameObjectWithTag("Player").GetComponent<HealthComponent>();
+                playerHealth.Death += OnPlayerDeath;
+
+                SetAllCanvasesToInactive();
+                break;
+            default:
+                Debug.Log("Scene names are wrong. Check MenuButtons script");   // TODO:: Remove when game is done. This is for future ease.
```
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs:105]:
```suggestion
                Debug.Log("Scene names are wrong. Check CanvasManager script");   // TODO:: Remove when game is done. This is for future ease.
```

File: aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs [hunk 4]
```diff
@@ -161,22 +161,55 @@ public void OnToggleSettings()
...
-        if (_currentSceneName == _sceneNameStartingMenu) // Toggle from menu to menu settings and back.
+        IsOnSettings = !IsOnSettings;
+
+        switch (_currentSceneName)
         {
-            IsOnMenuSettings = !IsOnMenuSettings;
-            SettingMenuCanvas.SetActive(IsOnMenuSettings);
-            MenuCanvas.SetActive(!IsOnMenuSettings);
+            case _sceneNameStartingMenu:
+                MenuCanvas.SetActive(!IsOnSettings);
```
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs:169]: Does this do what it is supposed to do? It is double negating, since you already flipped the bool before the switch case

File: aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs:1]: image

Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs:1]: It seems the return button does not always fit on the screen
image

**Review by JoenvandeVorle — CHANGES_REQUESTED** [review]
Looks nice, have a few comments. Nice use of switch statements!

File: aplib.net-demo/Assets/Art/UI/Background/HelpScreen.png
Comment (SSelinn) [reply aplib.net-demo/Assets/Art/UI/Background/HelpScreen.png:1]: ill add the jump key, but showing the cursor anytime is not a problem right?

Comment (SSelinn) [reply aplib.net-demo/Assets/Art/UI/Background/HelpScreen.png:1]: nvm, the player can jump.

File: aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs (hunk 1 above)
Comment (SSelinn) [reply aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs]: Readonly doesnt work with switch cases. I converted all of them to if statements and removed the modifiers.

File: aplib.net-demo/Assets/Art/UI/Background/HelpScreen.png
Comment (JoenvandeVorle) [reply aplib.net-demo/Assets/Art/UI/Background/HelpScreen.png:1]: it's not a problem no, but it should be added to the controls, like the WASD

File: aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs (hunk 2 above)
Comment (SSelinn) [reply aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs]: Added readonly's and changed the scene strings to above.

File: aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs (hunk 4 above)
Comment (SSelinn) [reply aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs]: On startup the bool IsOnSettings is on false. When the settings button is pressed the bool is true and sets all things that need to be on on. However, when going into the settings menu needs to be off.

tl;dr It does what it needs to do

File: aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs
Comment (SSelinn) [reply aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs:1]: This wont be a problem when the game scene string is GridSystem. This can be ignored

Comment (SSelinn) [reply aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs:1]: But does it need to be in free aspect ratio? When it is set to eg 16:9 it works

File: aplib.net-demo/Assets/Art/UI/Background/HelpScreen.png
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Art/UI/Background/HelpScreen.png:1]: I would still add the show mouse on left alt, and another important one is using items (control) and changing item (shift)!!

File: aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs [hunk 5]
```diff
@@ -117,12 +116,11 @@ private void Start()
...
         MenuCanvas.SetActive(false);
-        SettingMenuCanvas.SetActive(false);
-        SettingGameCanvas.SetActive(false);
+        SettingCanvas.SetActive(false);
         GameOverCanvas.SetActive(false);
         WinScreenCanvas.SetActive(false);
-        IsOnMenuSettings = false;
-        IsOnGameSettings = false;
+        HelpCanvas.SetActive(false);
+        IsOnSettings = false;
```
Comment (JensSteenmetz) [inline aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs:123]: Is this missing? I'm not sure.
```suggestion
        IsOnSettings = false;
        IsOnHelp = false;
```

**Review by JoenvandeVorle — DISMISSED** [review]

File: aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs [hunk 6]
```diff
@@ -40,13 +40,13 @@ public class CanvasManager : MonoBehaviour
     /// Name of the start screen.
     /// </summary>
     [SerializeField]
-    private string _sceneNameStartingMenu = "Settings"; // TODO:: Load main start screen
+    private readonly string _sceneNameStartingMenu = "Settings";
```
Comment (JensSteenmetz) [inline aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs:43]: The SerializeField is currently useless. If you don't want to make it visible in the inspector, remove the [SerializeField]. If you DO want to make it visible in the inspector (so that you can easily adjust the strings) it should not be readonly.

File: aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs [hunk 7]
```diff
@@ -40,13 +40,13 @@ public class CanvasManager : MonoBehaviour
...
 
     /// <summary>
     /// Name of the game scene
     /// </summary>
     [SerializeField]
```
Comment (JensSteenmetz) [inline aplib.net-demo/Assets/Scripts/MainMenu/CanvasManager.cs:48]: Same here.

**Review by JensSteenmetz — CHANGES_REQUESTED** [review]
Great work! A few comments

**Review by JensSteenmetz — APPROVED** [review]
LGTM!

**Review by JoenvandeVorle — APPROVED** [review]
bamigo
